infra/DockerRepoManager.py

```diff
@@ -0,0 +1,189 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Class to manage a git repos interaction with a docker image.
+
+This class is to be used to manage Oss-Fuzz projects docker build images.
+It allows for a build image to be build with a specific commit rather
+than just the current head of the repo.
+
+  Typical usage example:
+
+    drm = DockerRepoManager('curl')
+    current_commit = drm.get_image_commit()
+    drm.set_image_commit('df26f5f9c36e19cd503c0e462e9f72ad37b84c82')
+
+"""
+from helper import _build_image
+from helper import _check_project_exists
+from helper import _get_dockerfile_path
+from helper import _is_base_image
+from RepoManager import RepoManager
+import os
+
+
+class NoRepoFoundException(Exception):
+  """Occurs when the bisector cant infer the main repo."""
+  pass
+
+
+class DockerRepoManagerException(Exception):
+  """When there is a docker error of execution"""
+  pass
+
+
+class DockerRepoManager(RepoManager):
+  """Class to manage a git repo inside of a docker continer.
+
+  Attributes:
+    docker_image: The name of the docker image that is being modified
+    project_name: The name of the project associated with the docker image
+  """
+
+  docker_image = ''
+  project_name = ''
+  src_on_image = ''
+  TEMP_IMAGE_NAME = 'temp_container'
+
+  def __init__(self, project_name):
+    """Inits the DockerRepoManager class.
+
+      Args:
+        project_name: The name of the project that is associated with the image
+    """
+    self.project_name = project_name
+    self.docker_image = self._get_docker_image_name()
+    repo_url = self._infer_main_repo()
+    self.src_on_image = os.path.join('/src', project_name)
+    super().__init__(repo_url)
+    if self.repo_name != project_name:
+      raise DockerRepoManagerException(
+          'Error, the project name must be the same as the ' +
+          'git repo name but are %s and %s' % (project_name, super().repo_name))
+
+  def set_image_commit(self, commit):
+    """Creates a docker image with a specified commit as its source.
+
+    Args:
+      commit: The SHA the source is to be checked out at
+
+    Raises:
+      DockerRepoManagerException: when the commit is not successfully
+      mounted to the image
+    """
+
+    #Remove old temp container
+    stop_command = ['docker', 'stop', self.TEMP_IMAGE_NAME]
```

Comment:
lines 86-93 could be moved to a separate method named e.g. `cleanup` or somehow else